File: 485.MaxConsecutiveOnes/max.cc
#include <cmath>
using namespace std;
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxLen = 0;
        int len = 0;
        bool flag = false;

        for (int i = 0; i < int(nums.size()); i++) {
            if (nums[i] == 1) {
                if (flag == false) {
                    flag = true;
                }
                len++;
            }
            else {
                maxLen = max(maxLen, len);
                len = 0;
            }
        }
        return max(maxLen, len);
    }
};

File: 485.MaxConsecutiveOnes/max.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        flag = length = max = 0

        for i in range(len(nums)):
            if nums[i] == 1:
                if flag == 0:
                    flag = 1
                length += 1
            else:
                max = length > max and length or max
                length = 0
            max = length > max and length or max

        return max


if __name__ == '__main__':
    nums = [0]
    print(Solution().findMaxConsecutiveOnes(nums))
